lib/auth/auth.go

```diff
@@ -2947,22 +2950,60 @@ func (a *Server) GetAccessCapabilities(ctx context.Context, req types.AccessCapa
 	return caps, nil
 }
 
-// calculateMaxAccessTTL determines the maximum allowable TTL for a given access request
-// based on the MaxSessionTTLs of the roles being requested (a access request's life cannot
-// exceed the smallest allowable MaxSessionTTL value of the roles that it requests).
-func (a *Server) calculateMaxAccessTTL(ctx context.Context, req types.AccessRequest) (time.Duration, error) {
-	minTTL := defaults.MaxAccessDuration
-	for _, roleName := range req.GetRoles() {
+// resourceTTL calculates the TTL for the types.AccessRequest resource.
+func (a *Server) resourceTTL(ctx context.Context, identityExpires time.Time, r types.AccessRequest) (time.Duration, error) {
```

Comment:
If this is going to be a method on `*Server` I think it should have a more specific name like `accessRequestPendingTTL`, same for the other helpers.

But I think it may be more appropriate it implement this inside `RequestValidator.Validate`, we'd just need to pass in the clock and the identity expiry. Then we could try to keep all the interesting access request logic in one place and avoid adding even more helpers to `*Server`.